scss/_patterns_side-navigation.scss

```diff
@@ -28,13 +31,40 @@
     display: flex;
     padding: $spv-inner--x-small $sph-inner $spv-inner--x-small $sph-inner;
 
-    // nested 2 or 3 levels
+    // space for the right hand icon with default inner spacing on the left
+    $icon-spacing-width: map-get($icon-sizes, default) + $sph-inner;
+
+    .p-side-navigation--icons & {
+      padding-left: $sph-inner + $icon-spacing-width;
+      position: relative;
+    }
+
+    // nested 2nd level of navigation
     .p-side-navigation__item .p-side-navigation__item & {
       padding-left: 2 * $sph-inner;
```

Comment:
I think the levels should be configurable when you invoke the mixin; many sites won't need 3 levels, and some might need 5; my proposal: @mixin p-side-navigation($depth: 3)

Reply:
Feels a bit overkill.

The design was discussed many times and it seems for now we barely have examples that would use all 3 level of navigation. If anything really needs more nesting we could add it later.

Also nesting is just 2 not very long rules (4 precisely if you count `--icons` variant), so there is no much benefit in file size to reduce it to just one.

The complexity of adding configurable mixin that probably no one will use doesn't seem worth it. It doesn't seem like we do anything like this anywhere else. 

While flexibility and ability to configure things is great it comes with cost of maintenance and ease of use. 

I guess the biggest problem we are trying to solve with Vanilla is to keep our products consistent, so lets start with good default variants of components. If anything different is needed we can consider it on case by case basis.